cmake/CMakeLists.txt

```diff
@@ -917,28 +927,126 @@ endif()
 
 if (onnxruntime_ENABLE_TRAINING)
   add_compile_definitions(ENABLE_TRAINING)
+
+  if (UNIX)
+    # Find MPI
+    find_path(MPI_INCLUDE_DIR
+      NAMES mpi.h
+      HINTS
+      ${onnxruntime_MPI_HOME}/include
+      /bert_ort/openmpi/include)
+
+    set(MPI_LIBNAME "mpi")
+
+    find_library(MPI_LIBRARY
+      NAMES ${MPI_LIBNAME}
+      HINTS
+      ${onnxruntime_MPI_HOME}/lib
+      /bert_ort/openmpi/lib)
+
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(MPI DEFAULT_MSG MPI_INCLUDE_DIR MPI_LIBRARY)
+
+    if (MPI_FOUND)
+      set(MPI_HEADER_FILE "${MPI_INCLUDE_DIR}/mpi.h")
+      message( STATUS "Determining MPI version from the header file: ${MPI_HEADER_FILE}" )
+      file (STRINGS ${MPI_HEADER_FILE} MPI_MAJOR_VERSION_DEFINED
+            REGEX "^[ \t]*#define[ \t]+OMPI_MAJOR_VERSION[ \t]+[0-9]+.*$" LIMIT_COUNT 1)
+      if (MPI_MAJOR_VERSION_DEFINED)
+        string (REGEX REPLACE "^[ \t]*#define[ \t]+OMPI_MAJOR_VERSION[ \t]+" ""
+                MPI_MAJOR_VERSION ${MPI_MAJOR_VERSION_DEFINED})
+        message( STATUS "MPI_MAJOR_VERSION: ${MPI_MAJOR_VERSION}" )
+      endif()
+      file (STRINGS ${MPI_HEADER_FILE} MPI_MINOR_VERSION_DEFINED
+            REGEX "^[ \t]*#define[ \t]+OMPI_MINOR_VERSION[ \t]+[0-9]+.*$" LIMIT_COUNT 1)
+      if (MPI_MINOR_VERSION_DEFINED)
+        string (REGEX REPLACE "^[ \t]*#define[ \t]+OMPI_MINOR_VERSION[ \t]+" ""
+                MPI_MINOR_VERSION ${MPI_MINOR_VERSION_DEFINED})
+        message(STATUS "MPI_MINOR_VERSION: ${MPI_MINOR_VERSION}")
+      endif()
+
+      set(MPI_INCLUDE_DIRS ${MPI_INCLUDE_DIR})
+      set(MPI_LIBRARIES ${MPI_LIBRARY})
+      message( STATUS "Found MPI (include: ${MPI_INCLUDE_DIRS}, library: ${MPI_LIBRARIES})" )
+      mark_as_advanced(MPI_INCLUDE_DIRS MPI_LIBRARIES)
+
+      list(APPEND onnxruntime_EXTERNAL_LIBRARIES ${MPI_LIBRARIES} ${MPI_CXX_LINK_FLAGS})
+
+      add_definitions(-DUSE_MPI=1)
+      message( STATUS "MPI is enabled in Linux GPU Build." )
+    else ()
+      message( WARNING "MPI is not found. Please use --mpi_home to specify the path of MPI. Otherwise, MPI is disabled." )
+    endif()
+
+    # Find NCCL
+    if (onnxruntime_USE_CUDA)
+      find_path(NCCL_INCLUDE_DIR
+        NAMES nccl.h
+        HINTS
+        ${onnxruntime_NCCL_HOME}/include
+        $ENV{CUDA_ROOT}/include)
+
+      if ($ENV{USE_STATIC_NCCL})
```

Comment:
pytorch is using exactly the same env variable. Maybe prefix this with ort to avoid inference for those who develop pytorch and ort together.

Reply:
Just to confirm, are you saying USE_STATIC_NCCL?

Reply:
Yes.

Reply:
I am going to remove USE_STATIC_NCCL now since we don't use it now.